libkbfs/bserver_tlf_storage.go

```diff
@@ -0,0 +1,409 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+// TODO: Have bserverTlfStorage store a journal of put operations.
+
+// TODO: Do all high-level operations atomically on the file-system
+// level.
+
+// bserverTlfStorage stores block data for a single TLF in flat files
+// on disk. In particular, it stores data for each block in a
+// directory with name equal to the hex-encoded blockID.
+//
+// The block ID name is splayed over 256^2 subdirectories (one byte
```

Comment:
Given that there's unlikely to ever be 256 supported hashes, this number seems a bit large and confused me at first.  Maybe reword to make it clearer that currently there are only 256 possible directories?